app/src/main/java/com/gathering/android/signUp/SignUpActivity.kt

```diff
@@ -0,0 +1,82 @@
+import java.io.Serializable
+import javax.inject.Inject
+
+
+@AndroidEntryPoint
+class SignUpActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignUpBinding
+
+    @Inject
+    lateinit var viewModel: SignUpViewModel
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.etMail.doOnTextChanged { text, _, _, _ ->
+            viewModel.onEmailAddressChanged(text.toString())
+        }
+
+        binding.etPass.doOnTextChanged { text, _, _, _ ->
+            viewModel.onPasswordChanged(text.toString())
+        }
+
+        binding.etVerifyPass.doOnTextChanged { text, _, _, _ ->
+            val pass = binding.etPass.text.toString()
+            val confirmedPass = text.toString()
+            viewModel.onConfirmedPasswordChanged(pass, confirmedPass)
+        }
+
+        binding.btnSignUp.setOnClickListener {
+            val email = binding.etMail.text.toString()
+            val pass = binding.etPass.text.toString()
+            viewModel.onSignUpButtonClicked(email, pass)
+        }
+
+        viewModel.viewState.observe(this) { state ->
+            when (state) {
+                is SignUpScreenViewState.Error.ShowEmptyEmailError -> {
+                    binding.etMail.error = state.errorMessage
+                }
+                is SignUpScreenViewState.Error.ShowInvalidEmailError -> {
+                    binding.etMail.error = state.errorMessage
+                }
+                is SignUpScreenViewState.Error.ShowEmptyPassError -> {
+                    binding.etPass.error = state.errorMessage
+                }
+                is SignUpScreenViewState.Error.ShowInvalidPassError -> {
+                    binding.etPass.error = state.errorMessage
+                }
+                is SignUpScreenViewState.Error.ShowEmptyConfirmedPassError -> {
+                    binding.etVerifyPass.error = state.errorMessage
+                }
+                is SignUpScreenViewState.Error.ShowInvalidConfirmedPassError -> {
+                    binding.etVerifyPass.error = state.errorMessage
+                }
+                is SignUpScreenViewState.Error.ShowGeneralError -> {
+                }
+                is SignUpScreenViewState.NavigateToEventScreen -> {
+                    val intent = Intent(this, EventActivity::class.java)
+                    intent.putExtra("extra_object", state.user as Serializable)
+                    startActivity(intent)
+                }
+                is SignUpScreenViewState.SignUpButtonVisibility -> {
+                    binding.btnSignUp.isEnabled = state.isSignUpButtonEnabled
+                }
+                is SignUpScreenViewState.Error.ShowAuthenticationFailedError ->
+                    Toast.makeText(this, state.errorMessage, Toast.LENGTH_LONG).show()
```

Comment:
❗ You have a toast function here then you can create a private showToast() function that takes the text and then show a toast for both places. For general errors and authentication failure errors.

Reply:
I've created a private function for showing the toast :) thanks